### PR TITLE
use defaultLinkTarget when creating a link, fixes autolink

### PR DESCRIPTION
**Pull request by bettysteger**
fixes https://github.com/liferay/alloy-editor/issues/1534

### CODE REVIEW

File: src/core/link.js [hunk 1]
```diff
@@ -90,6 +90,14 @@ Link.prototype = {
 			attrs
 		);
 
+		if(!linkAttrs.target) {
+			const buttonCfg = this._editor.config.buttonCfg || {};
+			const defaultLinkTarget = buttonCfg['linkEdit'] && buttonCfg['linkEdit']['defaultLinkTarget'];
+			if(defaultLinkTarget) {
+				linkAttrs.target = defaultLinkTarget;
+			}
+		}
```
Comment (markocikos) [inline src/core/link.js:99]: `CKEDITOR.Link` is a low-level utility, for creating and handling links. `buttonCfg` is not an API intended to affect links in general, or link utilities. If is an API to configure specific button components. `linkEdit` is the key identifying [ButtonLinkEdit component](https://github.com/liferay/alloy-editor/blob/master/src/components/buttons/button-link-edit.jsx#L60). We should not couple `linkEdit` with links in general, like in this solution.

I could see this maybe as a new configuration option, on the level of `buttonCfg`. Something like
```js
linkCfg: {
    defaultTarget: '_blank'
}
```
Then, we could read it and merge with other attributes, in `link.js`.

Comment (bettysteger) [reply src/core/link.js]: @markocikos thanks for the proposal, i changed it to linkCfg.defaultTarget - makes total sense!

**Review by markocikos — CHANGES_REQUESTED** [review]
@bettysteger Looks good!

Before merging, please do two minor things:
1. Run `yarn format` on repo root, to format the code.
2. We started following [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) on this repo. Please squash and split code into two commits:
   - The first one, with core code modifications, should have a message starting with `feat: `.  It is not a fix, as we are adding new API.
   - The second one, with tests, should have a message starting with `test: `.

**Comment by markocikos** [comment]
see https://github.com/liferay/alloy-editor/pull/1536

**Pull request closed without merging**